src/Stratis.Bitcoin.IntegrationTests/Mempool/MempoolRelaySpecification_Steps.cs

```diff
@@ -0,0 +1,97 @@
+﻿using System.Linq;
+using FluentAssertions;
+using FluentAssertions.Common;
+using NBitcoin;
+using Stratis.Bitcoin.Connection;
+using Stratis.Bitcoin.Features.Consensus;
+using Stratis.Bitcoin.IntegrationTests.EnvironmentMockUpHelpers;
+using Xunit.Abstractions;
+
+namespace Stratis.Bitcoin.IntegrationTests.Mempool
+{
+    public partial class MempoolRelaySpecification
+    {
+        private NodeBuilder nodeBuilder;
+        private CoreNode nodeA;
+        private CoreNode nodeB;
+        private CoreNode nodeC;
+        private int coinbaseMaturity;
+        private Transaction transaction;
+
+        // NOTE: This constructor is allows test steps names to be logged
+        public MempoolRelaySpecification(ITestOutputHelper outputHelper) : base(outputHelper)
+        {
+        }
+
+        protected override void BeforeTest()
+        {
+            this.nodeBuilder = NodeBuilder.Create();
+        }
+
+        protected override void AfterTest()
+        {
+            this.nodeBuilder.Dispose();
+        }
+
+        protected void nodeA_nodeB_and_nodeC()
+        {
+            this.nodeA = this.nodeBuilder.CreateStratisPowNode();
+            this.nodeB = this.nodeBuilder.CreateStratisPowNode();
+            this.nodeC = this.nodeBuilder.CreateStratisPowNode();
+
+            this.nodeBuilder.StartAll();
+            this.nodeA.NotInIBD();
+            this.nodeB.NotInIBD();
+            this.nodeC.NotInIBD();
+
+            this.coinbaseMaturity = (int)this.nodeA.FullNode.Network.Consensus.Option<PowConsensusOptions>().CoinbaseMaturity;
+
+            // add some coins to nodeA
+            this.nodeA.SetDummyMinerSecret(new BitcoinSecret(new Key(), this.nodeA.FullNode.Network));
+            this.nodeA.GenerateStratisWithMiner(this.coinbaseMaturity + 1);
+            TestHelper.WaitLoop(() => TestHelper.IsNodeSynced(this.nodeA));
+        }
+
+        protected void nodeA_connects_to_nodeB()
+        {
+            this.nodeA.CreateRPCClient().AddNode(this.nodeB.Endpoint, true);
+            TestHelper.WaitLoop(() => TestHelper.AreNodesSynced(this.nodeA, this.nodeB));
+        }
+
+        protected void nodeA_nodeB_and_nodeC_are_not_whitelisted()
+        {
+            this.nodeA.FullNode.NodeService<IConnectionManager>().ConnectedPeers.First().Behavior<ConnectionManagerBehavior>().Whitelisted = false;
+            this.nodeB.FullNode.NodeService<IConnectionManager>().ConnectedPeers.First().Behavior<ConnectionManagerBehavior>().Whitelisted = false;
+            this.nodeC.FullNode.NodeService<IConnectionManager>().ConnectedPeers.First().Behavior<ConnectionManagerBehavior>().Whitelisted = false;
+        }
+
+        protected void nodeB_connects_to_nodeC()
+        {
+            this.nodeB.CreateRPCClient().AddNode(this.nodeC.Endpoint, true);
+            TestHelper.WaitLoop(() => TestHelper.AreNodesSynced(this.nodeA, this.nodeB));
```

Comment:
Did you mean node B and node C here in the AreNodesSynced?

Reply:
Good spot yes